internal/users/users.go

```diff
@@ -14,17 +14,46 @@ import (
 	"github.com/gorilla/mux"
 	"github.com/lib/pq"
 	"golang.org/x/crypto/bcrypt"
+	"gopkg.in/go-playground/validator.v9"
 	"net/http"
 	"net/smtp"
 	"net/url"
+	"strings"
 	"time"
 )
 
+type UserValid struct {
+	validate          *validator.Validate
+	failedValidations []string
+	validErr          map[string]string
+}
+type NameVal struct {
+	short    bool
+	long     bool
+	realName bool
```

Comment:
it's unclear what this means, I assume short is when the name is too short, and long is when the name is too long, and the `realName` is ... ?